src/solana/SolanaEngine.ts

```diff
@@ -775,13 +775,15 @@ export class SolanaEngine extends CurrencyEngine<
     })
     const versionedTx = new VersionedTransaction(versionedMessage)
 
-    const cleanSpendInfoOtherParams =
-      asSolanaSpendInfoOtherParams(spendInfoOtherParams)
+    let unsignedTx = versionedTx.serialize()
+    if (spendInfoOtherParams?.unsignedTx != null) {
+      const cleanSpendInfoOtherParams =
+        asSolanaTxOtherParams(spendInfoOtherParams)
+      unsignedTx = cleanSpendInfoOtherParams.unsignedTx
+    }
 
     const otherParams: JsonObject = {
```

Comment:
Can we do `const otherParams = wasSolanaTxOtherParams({ unsignedTx })`, with `const wasSolanaTxOtherParams = uncleaner(asSolanaTxOtherParams)`